### PR TITLE
PD-309: Card Component

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Created a card component and wrote tests to test the functionality. To note: this component is largely just a display component.

🎟 _Jira ticket:_ [PD-309](https://jira.mongodb.org/browse/PD-309)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/card/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  className='card-styles'
+>
+  This is my card component
+</Card>
+```
+
+**Output HTML**
+
+```HTML
+ <section class="leafygreen-ui-1xf4bcx">This is a card component</section>
```
Comment (DesignerDave) [inline packages/card/README.md]: Maybe `<article>` is a better semantic tag to use here. @hswolff Do you have a preference?

File: packages/card/src/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+
+const containerStyle = css`
+  background-color: white;
```
Comment (DesignerDave) [inline packages/card/src/Card.tsx]: Could ya add a transition to the border and box-shadow?

File: packages/card/src/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  }
+`;
+
+interface CardProps {
+  children: React.ReactNode;
+  as?: React.ElementType<any>;
+  className?: string;
+}
+
+function Card({ children, as = 'div', className, ...rest }: CardProps) {
```
Comment (DesignerDave) [inline packages/card/src/Card.tsx]: The default for `as` of `div` is inconsistent with your README (additionally we might want to default to something more semantic like `<article>`.

File: packages/card/src/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  const Root = as;
+  return (
```
Comment (DesignerDave) [inline packages/card/src/Card.tsx]: Line break above please

File: packages/card/README.md (hunk 1 above)
Comment (brookescarlett) [reply packages/card/README.md]: It's defaulting to div, section was just an example of using the as prop.

File: packages/card/src/Card.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/card/src/Card.tsx]: what kind of transition?

File: packages/card/src/Card.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/card/src/Card.tsx]:
```<Card
  as='section'
  className='card-styles'
>
  This is my card component
</Card>```

Comment (brookescarlett) [reply packages/card/src/Card.tsx]: just showing how an as work, can default to whatever you feel makes the most sense.

File: packages/card/src/Card.tsx (hunk 4 above)
Comment (brookescarlett) [reply packages/card/src/Card.tsx]: 👍

File: packages/card/README.md (hunk 1 above)
Comment (hswolff) [reply packages/card/README.md]: I don't have a strong preference.

I always feel weird using article as it's so semantically tied to news sites, which doesn't apply to most web apps.

Not sure which description on mdn applies best.

File: packages/card/src/Card.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  &:hover {
+    border: 1px solid ${uiColors.gray.light2};
+    box-shadow: 0px 3px 6px -2px ${uiColors.gray.base};
+  }
+`;
+
+interface CardProps {
+  children: React.ReactNode;
+  as?: React.ElementType<any>;
```
Comment (hswolff) [inline packages/card/src/Card.tsx]: I think TS has all the available DOM elements that we can include as the generic value.

Comment (brookescarlett) [reply packages/card/src/Card.tsx]: But what if someone wants to use as a <Link> element or something like that

Comment (brookescarlett) [reply packages/card/src/Card.tsx]: Hi @hswolff! Want to merge this in, but first wanted to make sure you were okay with leaving as-is. If you disagree, I'll update the type!

Comment (hswolff) [reply packages/card/src/Card.tsx]: Couldn't we have TS support both? 

Also what does it look like to have a Link as a card? That seems very odd.

**Pull request merged**